File: Ciftree/UI_NotesTasklist_SC.js
function UI_NotesTasklist_SC() {
    /*Scene:Include("UI_Notes_SC")
    Scene:Include("UI_Tasklist_SC")*/
    
    let ntMaster = AR.Transformer({
        collapsed: [0, 400],
        expanded: [0, 0],
        animTime: 0.2,
        z: 0,
        alias: "NotesTasklist"
    })
    let showTasklistHotspot = AR.Hotspot({
        ref: "UI_Frame_OVL",
        source: [13, 171, 122, 40],
        onScreen: [744, 242, 122, 40],
        cursor: "MenuHot",
        z: 3,
        active: false,
        id: "showTasklistHotspot",
        OnUp: ChangeTab.bind()
    });
    let showNotesHotspot = AR.Hotspot({
        ref: "UI_Frame_OVL",
        source: [175, 171, 122, 40],
        onScreen: [906, 242, 122, 40],
        cursor: "MenuHot",
        z: 3,
        active: true,
        id: "showNotesHotspot",
        OnUp: ChangeTab.bind()
    });
    let expandSound = AR.Sound({
        sounds: "TaskListOpen_SFX",
        channel: "UIFX",
        volume: 0.75,
        active: false
    });
    let collapseSound = AR.Sound({
        sounds: "TaskListClsd_SFX",
        channel: "UIFX",
        volume: 0.75,
        active: false
    });
    let expandTasklistButton = AR.Button({
        hs: AR.Hotspot({
            ref: "UI_Frame_OVL",
            source: [13, 299, 162, 40],
            onScreen: [744, 242, 162, 40],
            cursor: "MenuHot"
        }),
        overOvl: AR.Overlay({
          ovl: "UI_Frame_OVL",
          source: [969, 303, 994, 200],
          onScreen: [836, 254, 861, 279],
        }),
        z: 2,
        active: true,
        id: "expandTasklistButton",
        OnUp: Expand.bind(this) 
    });
    
    let collapseTasklistButton = AR.Button({
        hs: AR.Hotspot({
            ref: "UI_Frame_OVL",
            source: [13, 299, 162, 40],
            onScreen: [744, 242, 162, 40],
            cursor: "MenuHot"
        }),
        overOvl: AR.Overlay({
          ovl: "UI_Frame_OVL",
          source: [942, 303, 967, 200],
          onScreen: [836, 254, 861, 279],
        }),
        z: 2,
        active: false,
        id: "collapseTasklistButton",
        OnUp: Collapse.bind(this) 
    });
    
    ///
    
    let expandNotesButton = AR.Button({
        hs: AR.Hotspot({
            ref: "UI_Frame_OVL",
            source: [135, 299, 162, 40],
            onScreen: [866, 242, 162, 40],
            cursor: "MenuHot"
        }),
        overOvl: AR.Overlay({
          ovl: "UI_Frame_OVL",
          source: [969, 303, 994, 200],
          onScreen: [989, 252, 1014, 277]
        }),
        z: 3,
        active: false,
        id: "expandNotesButton",
        OnUp: Expand.bind(this) 
    });
    
    let collapseNotesButton = AR.Button({
        hs: AR.Hotspot({
            ref: "UI_Frame_OVL",
            source: [135, 299, 162, 40],
            onScreen: [866, 242, 162, 40],
            cursor: "MenuHot"
        }),
        overOvl: AR.Overlay({
          ovl: "UI_Frame_OVL",
          source: [942, 303, 967, 200],
          onScreen: [989, 252, 1014, 277],
        }),
        z: 3,
        active: false,
        id: "collapseNotesButton",
        OnUp: Collapse.bind(this) 
    });
    
    //let expandTasklistButton = AR.Button_obj(expandTasklistButton_obj);
    
    /*let resizeNotesButton = AR.Button({
        hs: AR.Hotspot({
          ref: "UI_Frame_OVL",
          source: [135, 171, 162, 40],
          onScreen: [866, 242, 162, 40],
          cursor: "MenuHot"
        }),
        overOvl: AR.Overlay({
          ovl: "UI_Frame_OVL",
          source: [969, 175, 994, 200],
          onScreen: [989, 252, 1014, 277]
        }),
        collapseSource: [942, 175, 967, 200],
        expandSource: [969, 175, 994, 200],
        z: 2,
        active: false,
        id: "resizeNotesButton"
    });*/
    
    let collapseButton = AR.Button({
        hs: AR.Hotspot({
          onScreen: [764, 729, 999, 759],
          cursor: "MenuHot"
        }),
        overOvl: AR.Overlay({
          ovl: "UI_Frame_OVL",
          source: [942, 303, 967, 200],
          onScreen: [876, 729, 901, 754]
        }),
        active: true,
        z: 2,
        id: "collapseButton",
        OnUp: Collapse.bind(this)
    });
    
    let expandedShownTasklistArrowOvl = AR.Overlay({
        ovl: "UI_Frame_OVL",
        source: [942, 276, 967, 173],
        onScreen: [836, 254, 861, 279],
        z: 1,
        active: false,
        id: "expandedShownTasklistArrowOvl"
    });
    let expandedShownNotesArrowOvl = AR.Overlay({
        ovl: "UI_Frame_OVL",
        source: [942, 276, 967, 173],
        onScreen: [989, 252, 1014, 277],
        z: 1,
        active: false,
        id: "expandedShownNotesArrowOvl"
    });
    let expandedHiddenTasklistArrowOvl = AR.Overlay({
        ovl: "UI_Frame_OVL",
        source: [969, 276, 994, 166],
        onScreen: [836, 252, 861, 270],
        z: 1,
        active: false,
        id: "expandedHiddenTasklistArrowOvl"
    });
    let expandedHiddenNotesArrowOvl = AR.Overlay({
        ovl: "UI_Frame_OVL",
        source: [969, 276, 994, 166],
        onScreen: [989, 250, 1014, 268],
        z: 1,
        active: false,
        id: "expandedHiddenNotesArrowOvl"
    });
    let backdropOvl = AR.Overlay({
        ovl: "UI_Frame_OVL",
        source: [686, 258, 940, 215],
        onScreen: [763, 676, 1017, 761],
        z: -4,
        active: true
    });
    let tasklistOvl = AR.Overlay({
        ovl: "UI_Frame_OVL",
        source: [2, 303, 297, 694],
        onScreen: [729, 242, 1024, 761],
        z: -1,
        active: true,
        id: "tasklistOvl"
    });
    let notesOVL = AR.Overlay({
        ovl: "UI_Frame_OVL",
        source: [299, 303, 594, 694],
        onScreen: [729, 242, 1024, 761],
        z: -1,
        active: false,
        id: "notesOVL"
    });
    /*let sink = AR.Hotspot({
        onScreen: tasklistOvl.onScreen,
        watchRight: true,
        cursor: "Menu",
        ref: tasklistOvl.ovl,
        source: tasklistOvl.source,
        z: -10
    });*/
    
    ntMaster.append(showTasklistHotspot, showNotesHotspot, expandTasklistButton, collapseTasklistButton, expandNotesButton, collapseNotesButton, collapseButton, expandedShownTasklistArrowOvl, expandedShownNotesArrowOvl, expandedHiddenTasklistArrowOvl, expandedHiddenNotesArrowOvl, tasklistOvl, notesOVL/*, sink, notesMaster, tasklistMaster*/);
    document.querySelector("#uiFrame").append(ntMaster, backdropOvl);
    
    let activeTab = 1;
    let desinationTab = 2;
    let tabsState = "Collapse";
    
    function Expand() {
        tabsState = "Expanded";
        expandSound.play();
    
        document.querySelector("#NotesTasklist").style.transform = "TranslateY(0)";
    
        switch (activeTab) {
            case 1:
                activatedTasklistExpand();
                deactivatedNotesExpand();
                break;
            case 2:
                activatedNotesExpand();
                deactivatedTasklistExpand();
                break;
        }
    }
    
    function Collapse() {
        tabsState = "Collapse";
        collapseSound.play();
    
        document.querySelector("#NotesTasklist").style.transform = "TranslateY(400px)";
    
        switch (activeTab) {
            case 1:
                activatedTasklistCollapse();
                deactivatedNotesCollapse();
                break;
            case 2:
                activatedNotesCollapse();
                deactivatedTasklistCollapse();
                break;
        }
    }
    
    function ChangeTab() {
        if(desinationTab == activeTab) return false;
        else {
            switch (desinationTab) {
                case 1:
                    activeTab = 1;
                    document.querySelector("#showTasklistHotspot").style.visibility = "hidden";
                    document.querySelector("#showNotesHotspot").style.visibility = "visible";
            
                    document.querySelector("#tasklistOvl").style.visibility = "visible";
                    document.querySelector("#notesOVL").style.visibility = "hidden";
            
                    if(tabsState == "Collapse") {
                        activatedTasklistCollapse();
                        deactivatedNotesCollapse();
                    }
                    else {
                        activatedTasklistExpand();
                        deactivatedNotesExpand();
                    }
                    
                    break;
                case 2:
                    activeTab = 2;
                    document.querySelector("#showTasklistHotspot").style.visibility = "visible";
                    document.querySelector("#showNotesHotspot").style.visibility = "hidden";
                
                    document.querySelector("#tasklistOvl").style.visibility = "hidden";
                    document.querySelector("#notesOVL").style.visibility = "visible";
                
                    if(tabsState == "Collapse") {
                        activatedNotesCollapse();
                        deactivatedTasklistCollapse();
                    }
                    else {
                        activatedNotesExpand();
                        deactivatedTasklistExpand();
                    }
                    
                    break;
            }
        }
        desinationTab = (activeTab == 1) ? 2 : 1;
    }
    
    /* TASKS */
    
    function activatedTasklistExpand() {
        document.querySelector("#collapseTasklistButton").style.visibility = "visible";
        document.querySelector("#expandTasklistButton").style.visibility = "hidden";
        document.querySelector("#expandedShownTasklistArrowOvl").style.visibility = "visible";
        document.querySelector("#expandedHiddenTasklistArrowOvl").style.visibility = "hidden";
    }
    
    function activatedTasklistCollapse() {
        document.querySelector("#collapseTasklistButton").style.visibility = "hidden";
        document.querySelector("#expandTasklistButton").style.visibility = "visible";
        document.querySelector("#expandedShownTasklistArrowOvl").style.visibility = "hidden";
        document.querySelector("#expandedHiddenTasklistArrowOvl").style.visibility = "hidden";
    }
    
    function deactivatedTasklistExpand() {
        document.querySelector("#collapseTasklistButton").style.visibility = "hidden";
        document.querySelector("#expandTasklistButton").style.visibility = "hidden";
        document.querySelector("#expandedShownTasklistArrowOvl").style.visibility = "hidden";
        document.querySelector("#expandedHiddenTasklistArrowOvl").style.visibility = "visible";
    }
    
    function deactivatedTasklistCollapse() {
        document.querySelector("#collapseTasklistButton").style.visibility = "hidden";
        document.querySelector("#expandTasklistButton").style.visibility = "hidden";
        document.querySelector("#expandedShownTasklistArrowOvl").style.visibility = "hidden";
        document.querySelector("#expandedHiddenTasklistArrowOvl").style.visibility = "hidden";
    }
    
    /* NOTES */
    
    function activatedNotesExpand() {
        document.querySelector("#collapseNotesButton").style.visibility = "visible";
        document.querySelector("#expandNotesButton").style.visibility = "hidden";
        document.querySelector("#expandedShownNotesArrowOvl").style.visibility = "visible";
        document.querySelector("#expandedHiddenNotesArrowOvl").style.visibility = "hidden";
    }
    
    function activatedNotesCollapse() {
        document.querySelector("#collapseNotesButton").style.visibility = "hidden";
        document.querySelector("#expandNotesButton").style.visibility = "visible";
        document.querySelector("#expandedShownNotesArrowOvl").style.visibility = "hidden";
        document.querySelector("#expandedHiddenNotesArrowOvl").style.visibility = "hidden";
    }
    
    function deactivatedNotesExpand() {
        document.querySelector("#collapseNotesButton").style.visibility = "hidden";
        document.querySelector("#expandNotesButton").style.visibility = "hidden";
        document.querySelector("#expandedShownNotesArrowOvl").style.visibility = "hidden";
        document.querySelector("#expandedHiddenNotesArrowOvl").style.visibility = "visible";
    }
    
    function deactivatedNotesCollapse() {
        document.querySelector("#collapseNotesButton").style.visibility = "hidden";
        document.querySelector("#expandNotesButton").style.visibility = "hidden";
        document.querySelector("#expandedShownNotesArrowOvl").style.visibility = "hidden";
        document.querySelector("#expandedHiddenNotesArrowOvl").style.visibility = "hidden";
    }
}

File: UI_NotesTasklist_SC.lua
Scene:Include("UI_Notes_SC")
Scene:Include("UI_Tasklist_SC")
local ntMaster = AR:Transformer({
  collapsed = Point:New(0, 400),
  expanded = Point:New(0, 0),
  animTime = 0.2,
  z = -3,
  alias = "NotesTasklist"
})
local showTasklistHotspot = AR:Hotspot({
  ref = "UI_Frame_OVL",
  source = Rect:LTWH(13, 171, 122, 40),
  onScreen = Rect:LTWH(744, 242, 122, 40),
  cursor = "MenuHot",
  active = false
})
local showNotesHotspot = AR:Hotspot({
  ref = "UI_Frame_OVL",
  source = Rect:LTWH(175, 171, 122, 40),
  onScreen = Rect:LTWH(906, 242, 122, 40),
  cursor = "MenuHot",
  active = false
})
local expandSound = AR:Sound({
  sounds = "TaskListOpen_SFX",
  channel = "UIFX",
  volume = 0.75,
  active = false
})
local collapseSound = AR:Sound({
  sounds = "TaskListClsd_SFX",
  channel = "UIFX",
  volume = 0.75,
  active = false
})
local resizeTasklistButton = AR:Button({
  hs = AR:Hotspot({
    ref = "UI_Frame_OVL",
    source = Rect:LTWH(13, 171, 162, 40),
    onScreen = Rect:LTWH(744, 242, 162, 40),
    cursor = "MenuHot"
  }),
  overOvl = AR:Overlay({
    ovl = "UI_Frame_OVL",
    source = Rect:New(969, 175, 994, 200),
    onScreen = Rect:New(836, 254, 861, 279)
  }),
  collapseSource = Rect:New(942, 175, 967, 200),
  expandSource = Rect:New(969, 175, 994, 200),
  z = 2,
  active = false
})
local resizeNotesButton = AR:Button({
  hs = AR:Hotspot({
    ref = "UI_Frame_OVL",
    source = Rect:LTWH(135, 171, 162, 40),
    onScreen = Rect:LTWH(866, 242, 162, 40),
    cursor = "MenuHot"
  }),
  overOvl = AR:Overlay({
    ovl = "UI_Frame_OVL",
    source = Rect:New(969, 175, 994, 200),
    onScreen = Rect:New(989, 252, 1014, 277)
  }),
  collapseSource = Rect:New(942, 175, 967, 200),
  expandSource = Rect:New(969, 175, 994, 200),
  z = 2,
  active = false
})
local collapseButton = AR:Button({
  hs = AR:Hotspot({
    onScreen = Rect:New(764, 729, 999, 759),
    cursor = "MenuHot"
  }),
  overOvl = AR:Overlay({
    ovl = "UI_Frame_OVL",
    source = Rect:New(942, 175, 967, 200),
    onScreen = Rect:New(876, 729, 901, 754)
  }),
  active = false
})
local expandedShownTasklistArrowOvl = AR:Overlay({
  ovl = "UI_Frame_OVL",
  source = Rect:New(942, 148, 967, 173),
  onScreen = Rect:New(836, 254, 861, 279),
  z = 1,
  active = false
})
local expandedShownNotesArrowOvl = AR:Overlay({
  ovl = "UI_Frame_OVL",
  source = Rect:New(942, 148, 967, 173),
  onScreen = Rect:New(989, 252, 1014, 277),
  z = 1,
  active = false
})
local expandedHiddenTasklistArrowOvl = AR:Overlay({
  ovl = "UI_Frame_OVL",
  source = Rect:New(969, 148, 994, 166),
  onScreen = Rect:New(836, 252, 861, 270),
  z = 1,
  active = false
})
local expandedHiddenNotesArrowOvl = AR:Overlay({
  ovl = "UI_Frame_OVL",
  source = Rect:New(969, 148, 994, 166),
  onScreen = Rect:New(989, 250, 1014, 268),
  z = 1,
  active = false
})
local backdropOvl = AR:Overlay({
  ovl = "UI_Frame_OVL",
  source = Rect:New(686, 130, 940, 215),
  onScreen = Rect:New(763, 676, 1017, 761),
  z = -4,
  active = true
})
local tasklistOvl = AR:Overlay({
  ovl = "UI_Frame_OVL",
  source = Rect:New(2, 175, 297, 694),
  onScreen = Rect:New(729, 242, 1024, 761),
  z = -1,
  active = false
})
local notesOVL = AR:Overlay({
  ovl = "UI_Frame_OVL",
  source = Rect:New(299, 175, 594, 694),
  onScreen = Rect:New(729, 242, 1024, 761),
  z = -1,
  active = false
})
local sink = AR:Hotspot({
  onScreen = tasklistOvl.onScreen,
  watchRight = true,
  cursor = "Menu",
  ref = tasklistOvl.ovl,
  source = tasklistOvl.source,
  z = -10
})
local activeARs = {}
local puzzleFlashOvl = AR:Overlay({
  ovl = "UI_Frame_OVL",
  source = Rect:New(686, 487, 935, 593),
  onScreen = Rect:New(764, 652, 1013, 758),
  active = false
})
local puzzleFlashTimer = AR:Timer({
  flash = false,
  duration = 0.33,
  flashCount = 0,
  flashMax = 3,
  active = false,
  RunOnce = function(this)
    if this.flash then
      puzzleFlashOvl.active = true
      this.flash = false
    else
      this.flash = true
    end
  end,
  OnDone = function(this)
    puzzleFlashOvl.active = false
    this.flashCount = this.flashCount + 1
    if this.flashCount <= this.flashMax and Flags.UI_PuzzleFlash_FL then
      this:Restart()
    end
  end
})
local function Expand(this)
  ntMaster.animator[1].t:AddBlendSlerp(ntMaster.animTime, ntMaster.expanded)
  ntMaster.animator[1].OnDone = nil
  ntMaster:SendAsAlias(ntMaster.alias, "NotesUI", "Expand")
  ntMaster:SendAsAlias(ntMaster.alias, "TasklistUI", "Expand")
  resizeTasklistButton.overOvl.source = resizeTasklistButton.collapseSource
  resizeNotesButton.overOvl.source = resizeNotesButton.collapseSource
  expandSound:Restart()
end
local function Collapse(this)
  ntMaster.animator[1].t:AddBlendSlerp(ntMaster.animTime, ntMaster.collapsed)
  ntMaster.animator[1].OnDone = function()
    ntMaster:SendAsAlias(ntMaster.alias, "NotesUI", "Collapse")
    ntMaster:SendAsAlias(ntMaster.alias, "TasklistUI", "Collapse")
  end
  resizeTasklistButton.overOvl.source = resizeTasklistButton.expandSource
  resizeNotesButton.overOvl.source = resizeNotesButton.expandSource
  collapseSound:Restart()
end
local function ShowPuzzleHelp(this)
  ntMaster:SendAsAlias(ntMaster.alias, "NotesUI", "ScrollToStart")
  puzzleFlashTimer.flashCount = 0
  puzzleFlashTimer.flash = false
  puzzleFlashTimer:Restart()
end
local function Activate(...)
  for _, ar in ipairs({
    ...
  }) do
    if ar.done then
      ar:Restart()
    else
      ar.active = true
    end
    table.insert(activeARs, ar)
  end
end
local function Deactivate()
  for _, ar in ipairs(activeARs) do
    ar.active = false
  end
  activeARs = {}
end
if not ntMaster:RegisterAlias(ntMaster.alias) then
  error("NotesTasklist could not register its alias")
end
ntMaster.animator[1].t = ntMaster.collapsed
ntMaster:Attach(showTasklistHotspot, showNotesHotspot, resizeTasklistButton, resizeNotesButton, collapseButton, expandedShownTasklistArrowOvl, expandedShownNotesArrowOvl, expandedHiddenTasklistArrowOvl, expandedHiddenNotesArrowOvl, tasklistOvl, notesOVL, sink, notesMaster, tasklistMaster)
uiFrame:Attach(ntMaster, backdropOvl)
function showTasklistHotspot.OnDone(this)
  EventManager:Raise("NT", "ShowTasklist")
end
function showNotesHotspot.OnDone(this)
  EventManager:Raise("NT", "ShowNotes")
end
function resizeTasklistButton.OnUp(this)
  EventManager:Raise("NT", "ResizeTasklist")
end
function resizeNotesButton.OnUp(this)
  EventManager:Raise("NT", "ResizeNotes")
end
function sink.OnDone(this)
  this:Reset()
end
function collapseButton.OnUp()
  EventManager:Raise("NT", "Collapse")
end
function ntMaster.Receive(this, sender, action)
  if sender == "NotesUI" and action == "Expand" then
    EventManager:Raise("NT", "ExpandNotes")
  end
  if string.comparenocase(action, "TutorialExpand") then
    EventManager:Raise("NT", "ResizeTasklist")
  end
  if string.comparenocase(action, "Collapse") then
    EventManager:Raise("NT", "Collapse")
  end
  if string.comparenocase(action, "ShowPuzzleHelp") then
    EventManager:Raise("NT", "ShowPuzzleHelp")
  end
  if string.comparenocase(action, "ShowTasklist") then
    EventManager:Raise("NT", "ShowTasklist")
  end
end
Scene:Include("Actor_SC")
local ntActor = MakeActor({name = "NT"})
ntActor:AddState({
  name = "collapsedTasklist",
  OnBegin = function(this)
    Activate(showNotesHotspot, resizeTasklistButton, tasklistOvl, tasklistMaster)
  end,
  OnEvent("NT", "ShowNotes", function(this)
    this.state = "collapsedNotes"
  end),
  OnEvent("NT", "ResizeTasklist", function(this)
    Expand()
    this.state = "expandedTasklist"
    Flags.UI_TasklistExpanded_FL = true
  end),
  OnEvent("NT", "ShowPuzzleHelp", function(this)
    ShowPuzzleHelp()
    this.state = "collapsedNotes"
  end),
  OnEnd = function(this)
    Deactivate()
  end
})
ntActor:AddState({
  name = "collapsedNotes",
  OnBegin = function(this)
    Activate(showTasklistHotspot, resizeNotesButton, notesOVL, notesMaster)
  end,
  OnEvent("NT", "ShowTasklist", function(this)
    this.state = "collapsedTasklist"
  end),
  OnEvent("NT", "ResizeNotes", function(this)
    Expand()
    this.state = "expandedNotes"
  end),
  OnEvent("NT", "ExpandNotes", function(this)
    Expand()
    this.state = "expandedNotes"
  end),
  OnEvent("NT", "ShowPuzzleHelp", function(this)
    ShowPuzzleHelp()
  end),
  OnEnd = function(this)
    Deactivate()
  end
})
ntActor:AddState({
  name = "expandedTasklist",
  OnBegin = function(this)
    Activate(showNotesHotspot, resizeTasklistButton, expandedShownTasklistArrowOvl, expandedHiddenNotesArrowOvl, collapseButton, tasklistOvl, tasklistMaster)
  end,
  OnEvent("NT", "ShowNotes", function(this)
    this.state = "expandedNotes"
    Flags.UI_TasklistExpanded_FL = false
  end),
  OnEvent("NT", "ResizeTasklist", function(this)
    Collapse()
    this.state = "collapsedTasklist"
    Flags.UI_TasklistExpanded_FL = false
  end),
  OnEvent("NT", "Collapse", function(this)
    EventManager:Raise("NT", "ResizeTasklist")
  end),
  OnEvent("NT", "ShowPuzzleHelp", function(this)
    Collapse()
    ShowPuzzleHelp()
    this.state = "collapsedNotes"
    Flags.UI_TasklistExpanded_FL = false
  end),
  OnEnd = function(this)
    Deactivate()
  end
})
ntActor:AddState({
  name = "expandedNotes",
  OnBegin = function(this)
    Activate(showTasklistHotspot, resizeNotesButton, expandedShownNotesArrowOvl, expandedHiddenTasklistArrowOvl, collapseButton, notesOVL, notesMaster)
  end,
  OnEvent("NT", "ShowTasklist", function(this)
    this.state = "expandedTasklist"
    Flags.UI_TasklistExpanded_FL = true
  end),
  OnEvent("NT", "ResizeNotes", function(this)
    Collapse()
    this.state = "collapsedNotes"
  end),
  OnEvent("NT", "Collapse", function(this)
    EventManager:Raise("NT", "ResizeNotes")
  end),
  OnEvent("NT", "ShowPuzzleHelp", function(this)
    Collapse()
    ShowPuzzleHelp()
    this.state = "collapsedNotes"
  end),
  OnEnd = function(this)
    Deactivate()
  end
})
if Flags.UI_PuzzleFlash_FL then
  ntActor.state = "collapsedNotes"
else
  ntActor.state = "collapsedTasklist"
end
